eng/targets/Npm.Common.targets

```diff
@@ -31,8 +36,15 @@
              BuildInParallel="true" />
   </Target>
 
-  <Target Name="Build" DependsOnTargets="PrepareForBuild;ResolveProjectReferences">
-    <Yarn Command="run build" Condition="'$(IsBuildable)' != 'false'" StandardOutputImportance="High" StandardErrorImportance="High" />
+  <Target Name="Build" DependsOnTargets="PrepareForBuild;ResolveProjectReferences;_Build" >
+  </Target>
+
+  <Target Name="_Build"
+    Condition="'$(IsBuildable)' != 'false'"
+    Inputs="@(TSFiles)"
+    Outputs="$(MSBuildProjectDirectory)\build-sentinel" >
+    <Yarn Command="run build" StandardOutputImportance="High" StandardErrorImportance="High" />
+    <WriteLinesToFile File="$(MSBuildProjectDirectory)\build-sentinel" />
```

Comment:
Suggest adding `Overwrite="true"` to update the creation as well as the modification timestamps

Reply:
Suggest placing this into `$(BaseIntermediateOutputPath)`, ensuring that a clean op will remove the file